From 360be327f9a03d01100ef47a71aba119542906d1 Mon Sep 17 00:00:00 2001
From: Stanislav Asunkin <1353637+stasjok@users.noreply.github.com>
Date: Thu, 23 Oct 2025 20:34:44 +0300
Subject: perf: do not index included documents

This language server is already indexing all beancount files after
startup. No need to index included documents since they should already
be indexed.

diff --git a/packages/lsp-server/src/common/features/symbol-index.ts b/packages/lsp-server/src/common/features/symbol-index.ts
index 6c96f9c..62610d4 100644
--- a/packages/lsp-server/src/common/features/symbol-index.ts
+++ b/packages/lsp-server/src/common/features/symbol-index.ts
@@ -1,4 +1,3 @@
-import mm from 'micromatch';
 import { CancellationTokenSource } from 'vscode-languageserver';
 import { TextDocument } from 'vscode-languageserver-textdocument';
 import { isInteresting, parallel, StopWatch } from '../common';
@@ -13,7 +12,6 @@ import {
 } from './symbol-extractors';
 
 import { Logger } from '@bean-lsp/shared';
-import { URI, Utils as UriUtils } from 'vscode-uri';
 import { TreeQuery } from '../language';
 import { BeancountOptionsManager, SupportedOption } from '../utils/beancount-options';
 import { globalEventBus, GlobalEvents } from '../utils/event-bus';
@@ -192,33 +190,6 @@ export class SymbolIndex {
 
 		this._symbolInfoStorage.removeSync({ _uri: document.uri });
 		await this._symbolInfoStorage.insertAsync(symbols)
-
-		await yieldToMain();
-		const tree = await this._trees.getParseTree(document);
-		if (tree) {
-			const captures = await TreeQuery.captures('(include (string) @path)', tree.rootNode);
-			const includePatterns = captures.map((c: { node: { text: string } }) => {
-				const text = c.node.text;
-				const stripedQuotationMark = text.replace(/^"/, '').replace(/"$/, '');
-				const u = UriUtils.joinPath(UriUtils.dirname(URI.parse(document.uri)), stripedQuotationMark);
-				return u.path;
-			});
-			let hasNew = false;
-			const beanFiles = this._documents.beanFiles;
-			this.logger.debug(`[index] Found ${beanFiles.length} bean files`);
-			includePatterns.forEach((pattern: string) => {
-				const list = beanFiles;
-				const matched = mm.match(list, pattern, { contains: true });
-				matched.map((p: string) => p).forEach((uri: string) => {
-					hasNew = true;
-					this.addAsyncFile(uri);
-				});
-			});
-			if (hasNew) {
-				await yieldToMain();
-				this.unleashFiles([]);
-			}
-		}
 	}
 
 	/**
-- 
2.50.1

